### PR TITLE
Update all AVM tests for post-linearization

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

The AVM tests were pretty bad.

- [X] Updates all of the tests to use a common testing setup.
- [X] Replaces issuance of transactions into vertices with issuance of transactions into blocks.
- [X] Removes duplicate tests.

## How this works

Total re-write of the testing setup.

## How this was tested

CI

### CODE REVIEW

File: vms/avm/service_test.go [hunk 1]
```diff
@@ -672,27 +510,25 @@ func TestServiceGetTxJSON_BaseTx(t *testing.T) {
...
-	require.Contains(jsonString, "\"outputs\":[{\"assetID\":\"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ\",\"fxID\":\"11111111111111111111111111111111LpoYY\",\"output\":{\"addresses\":[\"X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e\"],\"amount\":49000,\"locktime\":0,\"threshold\":1}}]")
+	require.Contains(jsonString, `"memo":"0x0102030405060708"`)
+	require.Contains(jsonString, `"inputs":[{"txID":"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ","outputIndex":2,"assetID":"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ","fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","input":{"amount":50000,"signatureIndices":[0]}}]`)
+	require.Contains(jsonString, `"outputs":[{"assetID":"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ","fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","output":{"addresses":["X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e"],"amount":49000,"locktime":0,"threshold":1}}]`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:515]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/service_test.go [hunk 2]
```diff
@@ -701,63 +537,30 @@ func TestServiceGetTxJSON_ExportTx(t *testing.T) {
...
-	require.Contains(jsonString, "\"inputs\":[{\"txID\":\"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ\",\"outputIndex\":2,\"assetID\":\"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ\",\"fxID\":\"11111111111111111111111111111111LpoYY\",\"input\":{\"amount\":50000,\"signatureIndices\":[0]}}]")
-	require.Contains(jsonString, "\"exportedOutputs\":[{\"assetID\":\"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ\",\"fxID\":\"11111111111111111111111111111111LpoYY\",\"output\":{\"addresses\":[\"X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e\"],\"amount\":49000,\"locktime\":0,\"threshold\":1}}]}")
+	require.Contains(jsonString, `"inputs":[{"txID":"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ","outputIndex":2,"assetID":"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ","fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","input":{"amount":50000,"signatureIndices":[0]}}]`)
+	require.Contains(jsonString, `"exportedOutputs":[{"assetID":"2XGxUr7VF7j1iwUp2aiGe4b6Ue2yyNghNS1SuNTNmZ77dPpXFZ","fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","output":{"addresses":["X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e"],"amount":49000,"locktime":0,"threshold":1}}]}`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:541]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/service_test.go [hunk 3]
```diff
@@ -768,67 +571,35 @@ func TestServiceGetTxJSON_CreateAssetTx(t *testing.T) {
...
+	require.Contains(jsonString, `"initialStates":[{"fxIndex":0,"fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","outputs":[{"addresses":["X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e"],"locktime":0,"threshold":1},{"addresses":["X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e"],"locktime":0,"threshold":1}]},{"fxIndex":1,"fxID":"qd2U4HDWUvMrVUeTcCHp6xH3Qpnn1XbU5MDdnBoiifFqvgXwT","outputs":[{"addresses":["X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e"],"groupID":1,"locktime":0,"threshold":1},{"addresses":["X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e"],"groupID":2,"locktime":0,"threshold":1}]},{"fxIndex":2,"fxID":"rXJsCSEYXg2TehWxCEEGj6JU2PWKTkd6cBdNLjoe2SpsKD9cy","outputs":[{"addresses":["X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e"],"locktime":0,"threshold":1},{"addresses":["X-testing1lnk637g0edwnqc2tn8tel39652fswa3xk4r65e"],"locktime":0,"threshold":1}]}]},"credentials":[],"id":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS"}`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:575]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/service_test.go [hunk 4]
```diff
@@ -838,75 +609,43 @@ func TestServiceGetTxJSON_OperationTxWithNftxMintOp(t *testing.T) {
...
 	// contains the fxID
-	require.Contains(jsonString, "\"operations\":[{\"assetID\":\"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS\",\"inputIDs\":[{\"txID\":\"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS\",\"outputIndex\":2}],\"fxID\":\"TtF4d2QWbk5vzQGTEPrN48x6vwgAoAmKQ9cbp79inpQmcRKES\"")
-	require.Contains(jsonString, "\"credentials\":[{\"fxID\":\"TtF4d2QWbk5vzQGTEPrN48x6vwgAoAmKQ9cbp79inpQmcRKES\",\"credential\":{\"signatures\":[\"0x571f18cfdb254263ab6b987f742409bd5403eafe08b4dbc297c5cd8d1c85eb8812e4541e11d3dc692cd14b5f4bccc1835ec001df6d8935ce881caf97017c2a4801\"]}}]")
+	require.Contains(jsonString, `"operations":[{"assetID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","inputIDs":[{"txID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","outputIndex":2}],"fxID":"qd2U4HDWUvMrVUeTcCHp6xH3Qpnn1XbU5MDdnBoiifFqvgXwT"`)
+	require.Contains(jsonString, `"credentials":[{"fxID":"qd2U4HDWUvMrVUeTcCHp6xH3Qpnn1XbU5MDdnBoiifFqvgXwT","credential":{"signatures":["0x571f18cfdb254263ab6b987f742409bd5403eafe08b4dbc297c5cd8d1c85eb8812e4541e11d3dc692cd14b5f4bccc1835ec001df6d8935ce881caf97017c2a4801"]}}]`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:618]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/service_test.go [hunk 5]
```diff
@@ -917,70 +656,38 @@ func TestServiceGetTxJSON_OperationTxWithMultipleNftxMintOp(t *testing.T) {
...
+	require.Contains(jsonString, `"operations":[{"assetID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","inputIDs":[{"txID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","outputIndex":2}],"fxID":"qd2U4HDWUvMrVUeTcCHp6xH3Qpnn1XbU5MDdnBoiifFqvgXwT"`)
+	require.Contains(jsonString, `"credentials":[{"fxID":"qd2U4HDWUvMrVUeTcCHp6xH3Qpnn1XbU5MDdnBoiifFqvgXwT","credential":{"signatures":["0x2400cf2cf978697b3484d5340609b524eb9dfa401e5b2bd5d1bc6cee2a6b1ae41926550f00ae0651c312c35e225cb3f39b506d96c5170fb38a820dcfed11ccd801"]}},{"fxID":"qd2U4HDWUvMrVUeTcCHp6xH3Qpnn1XbU5MDdnBoiifFqvgXwT","credential":{"signatures":["0x2400cf2cf978697b3484d5340609b524eb9dfa401e5b2bd5d1bc6cee2a6b1ae41926550f00ae0651c312c35e225cb3f39b506d96c5170fb38a820dcfed11ccd801"]}}]`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:663]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/service_test.go [hunk 6]
```diff
@@ -991,76 +698,44 @@ func TestServiceGetTxJSON_OperationTxWithSecpMintOp(t *testing.T) {
...
 
 	// contains the fxID
-	require.Contains(jsonString, "\"operations\":[{\"assetID\":\"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS\",\"inputIDs\":[{\"txID\":\"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS\",\"outputIndex\":0}],\"fxID\":\"LUC1cmcxnfNR9LdkACS2ccGKLEK7SYqB4gLLTycQfg1koyfSq\"")
-	require.Contains(jsonString, "\"credentials\":[{\"fxID\":\"LUC1cmcxnfNR9LdkACS2ccGKLEK7SYqB4gLLTycQfg1koyfSq\",\"credential\":{\"signatures\":[\"0x6d7406d5e1bdb1d80de542e276e2d162b0497d0df1170bec72b14d40e84ecf7929cb571211d60149404413a9342fdfa0a2b5d07b48e6f3eaea1e2f9f183b480500\"]}}]")
+	require.Contains(jsonString, `"operations":[{"assetID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","inputIDs":[{"txID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","outputIndex":0}],"fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ"`)
+	require.Contains(jsonString, `"credentials":[{"fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","credential":{"signatures":["0x6d7406d5e1bdb1d80de542e276e2d162b0497d0df1170bec72b14d40e84ecf7929cb571211d60149404413a9342fdfa0a2b5d07b48e6f3eaea1e2f9f183b480500"]}}]`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:708]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/service_test.go [hunk 7]
```diff
@@ -1071,71 +746,39 @@ func TestServiceGetTxJSON_OperationTxWithMultipleSecpMintOp(t *testing.T) {
...
+	require.Contains(jsonString, `"assetID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","inputIDs":[{"txID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","outputIndex":1}],"fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ"`)
+	require.Contains(jsonString, `"credentials":[{"fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","credential":{"signatures":["0xcc650f48341601c348d8634e8d207e07ea7b4ee4fbdeed3055fa1f1e4f4e27556d25056447a3bd5d949e5f1cbb0155bb20216ac3a4055356e3c82dca74323e7401"]}},{"fxID":"spdxUxVJQbX85MGxMHbKw1sHxMnSqJ3QBzDyDYEP3h6TLuxqQ","credential":{"signatures":["0xcc650f48341601c348d8634e8d207e07ea7b4ee4fbdeed3055fa1f1e4f4e27556d25056447a3bd5d949e5f1cbb0155bb20216ac3a4055356e3c82dca74323e7401"]}}]`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:754]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/service_test.go [hunk 8]
```diff
@@ -1146,75 +789,43 @@ func TestServiceGetTxJSON_OperationTxWithPropertyFxMintOp(t *testing.T) {
...
 
 	// contains the fxID
-	require.Contains(jsonString, "\"assetID\":\"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS\",\"inputIDs\":[{\"txID\":\"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS\",\"outputIndex\":4}],\"fxID\":\"2mcwQKiD8VEspmMJpL1dc7okQQ5dDVAWeCBZ7FWBFAbxpv3t7w\"")
-	require.Contains(jsonString, "\"credentials\":[{\"fxID\":\"2mcwQKiD8VEspmMJpL1dc7okQQ5dDVAWeCBZ7FWBFAbxpv3t7w\",\"credential\":{\"signatures\":[\"0xa3a00a03d3f1551ff696d6c0abdde73ae7002cd6dcce1c37d720de3b7ed80757411c9698cd9681a0fa55ca685904ca87056a3b8abc858a8ac08f45483b32a80201\"]}}]")
+	require.Contains(jsonString, `"assetID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","inputIDs":[{"txID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","outputIndex":4}],"fxID":"rXJsCSEYXg2TehWxCEEGj6JU2PWKTkd6cBdNLjoe2SpsKD9cy"`)
+	require.Contains(jsonString, `"credentials":[{"fxID":"rXJsCSEYXg2TehWxCEEGj6JU2PWKTkd6cBdNLjoe2SpsKD9cy","credential":{"signatures":["0xa3a00a03d3f1551ff696d6c0abdde73ae7002cd6dcce1c37d720de3b7ed80757411c9698cd9681a0fa55ca685904ca87056a3b8abc858a8ac08f45483b32a80201"]}}]`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:798]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/service_test.go [hunk 9]
```diff
@@ -1225,23 +836,23 @@ func TestServiceGetTxJSON_OperationTxWithPropertyFxMintOpMultiple(t *testing.T)
...
+	require.Contains(jsonString, `"operations":[{"assetID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","inputIDs":[{"txID":"2MDgrsBHMRsEPa4D4NA1Bo1pjkVLUK173S3dd9BgT2nCJNiDuS","outputIndex":4}],"fxID":"rXJsCSEYXg2TehWxCEEGj6JU2PWKTkd6cBdNLjoe2SpsKD9cy"`)
+	require.Contains(jsonString, `"credentials":[{"fxID":"rXJsCSEYXg2TehWxCEEGj6JU2PWKTkd6cBdNLjoe2SpsKD9cy","credential":{"signatures":["0x25b7ca14df108d4a32877bda4f10d84eda6d653c620f4c8d124265bdcf0ac91f45712b58b33f4b62a19698325a3c89adff214b77f772d9f311742860039abb5601"]}},{"fxID":"rXJsCSEYXg2TehWxCEEGj6JU2PWKTkd6cBdNLjoe2SpsKD9cy","credential":{"signatures":["0x25b7ca14df108d4a32877bda4f10d84eda6d653c620f4c8d124265bdcf0ac91f45712b58b33f4b62a19698325a3c89adff214b77f772d9f311742860039abb5601"]}}]`)
```
Comment (StephenButtolph) [inline vms/avm/service_test.go:843]: Only changes here should be the `fxID`s because we use the real `fxID`s now.

File: vms/avm/index_test.go [hunk 10]
```diff
@@ -5,349 +5,177 @@ package avm
...
-		for _, utxoID := range uniqueParsedTX.InputUTXOs() {
-			utxo, err := vm.state.GetUTXO(utxoID.InputID())
-			require.NoError(err)
-
-			inputUTXOs = append(inputUTXOs, utxo)
-		}
+		issueAndAccept(require, env.vm, env.issuer, tx)
 
-		// index the transaction
-		require.NoError(vm.addressTxsIndexer.Accept(uniqueParsedTX.ID(), inputUTXOs, uniqueParsedTX.UTXOs()))
```
Comment (StephenButtolph) [inline vms/avm/index_test.go:104]: I have no clue why we didn't actually `Accept` the transaction previously to test the indexing flow... instead we manually indexed it... Now we just accept it normally.

File: vms/avm/index_test.go [hunk 11]
```diff
@@ -5,349 +5,177 @@ package avm
...
```
Comment (StephenButtolph) [inline vms/avm/index_test.go:180]: Similarly to above, we now just accept the tx rather than manually indexing it.

File: vms/avm/index_test.go [hunk 12]
```diff
@@ -5,349 +5,177 @@ package avm
...
+	issueAndAccept(require, env.vm, env.issuer, tx)
 
-	// index the transaction
-	require.NoError(vm.addressTxsIndexer.Accept(tx.ID(), inputUTXOs, tx.UTXOs()))
-
-	assertIndexedTX(t, vm.db, uint64(0), addr, txAssetID.ID, tx.ID())
-	assertLatestIdx(t, vm.db, addr, txAssetID.ID, 1)
-}
-
-func TestIndexTransaction_UnorderedWrites(t *testing.T) {
```
Comment (StephenButtolph) [inline vms/avm/index_test.go:251]: This test was an **_exact_** copy of `TestIndexTransaction_MultipleTransactions` (except for a comment and whitespace)

File: vms/avm/index_test.go [hunk 13]
```diff
@@ -358,13 +186,10 @@ func TestIndexer_Read(t *testing.T) {
 func TestIndexingNewInitWithIndexingEnabled(t *testing.T) {
 	require := require.New(t)
 
-	baseDBManager := manager.NewMemDB(version.Semantic1_0_0)
-	ctx := NewContext(t)
-
-	db := baseDBManager.NewPrefixDBManager([]byte{1}).Current().Database
```
Comment (StephenButtolph) [inline vms/avm/index_test.go:364]: There is no reason to use the db manager here... No clue why it ever was.

File: vms/avm/index_test.go [hunk 14]
```diff
@@ -406,24 +230,21 @@ func TestIndexingNewInitWithIndexingDisabled(t *testing.T) {
 func TestIndexingAllowIncomplete(t *testing.T) {
 	require := require.New(t)
 
-	baseDBManager := manager.NewMemDB(version.Semantic1_0_0)
-	ctx := NewContext(t)
-
-	prefixDB := baseDBManager.NewPrefixDBManager([]byte{1}).Current().Database
```
Comment (StephenButtolph) [inline vms/avm/index_test.go:412]: There is no reason to use the db manager here... No clue why it ever was.

File: vms/avm/index_test.go [hunk 15]
```diff
@@ -437,10 +258,6 @@ func buildPlatformUTXO(utxoID avax.UTXOID, txAssetID avax.Asset, addr ids.ShortI
 	}
 }
 
-func signTX(codec codec.Manager, tx *txs.Tx, key *secp256k1.PrivateKey) error {
-	return tx.SignSECP256K1Fx(codec, [][]*secp256k1.PrivateKey{{key}})
-}
```
Comment (StephenButtolph) [inline vms/avm/index_test.go:442]: This was just inlined where it was called before... It's a one-liner either way.

File: vms/avm/index_test.go [hunk 16]
```diff
@@ -468,74 +285,28 @@ func buildTX(utxoID avax.UTXOID, txAssetID avax.Asset, address ...ids.ShortID) *
...
 func assertLatestIdx(t *testing.T, db database.Database, sourceAddress ids.ShortID, assetID ids.ID, expectedIdx uint64) {
 	require := require.New(t)
 
 	addressDB := prefixdb.New(sourceAddress[:], db)
 	assetDB := prefixdb.New(assetID[:], addressDB)
 
-	expectedIdxBytes := make([]byte, wrappers.LongLen)
-	binary.BigEndian.PutUint64(expectedIdxBytes, expectedIdx)
-
+	expectedIdxBytes := database.PackUInt64(expectedIdx)
```
Comment (StephenButtolph) [inline vms/avm/index_test.go:294]: We should probably make this change in the actual indexing code as well - but I didn't want to expand the real code changes.

File: vms/avm/index_test.go [hunk 17]
```diff
@@ -468,74 +285,28 @@ func buildTX(utxoID avax.UTXOID, txAssetID avax.Asset, address ...ids.ShortID) *
...
-	return nil
-}
-
-func assertIndexedTX(t *testing.T, db database.Database, index uint64, sourceAddress ids.ShortID, assetID ids.ID, transactionID ids.ID) {
-	require.NoError(t, checkIndexedTX(db, index, sourceAddress, assetID, transactionID))
+	idxBytes := database.PackUInt64(index)
+	txID, err := database.GetID(assetDB, idxBytes)
+	require.NoError(err)
+	require.Equal(transactionID, txID)
 }
```
Comment (StephenButtolph) [inline vms/avm/index_test.go:310]: I have no clue why these were two different functions...

File: vms/avm/state_test.go
Comment (StephenButtolph) [inline vms/avm/state_test.go:1]: I question the usefulness of these tests... (I think they are completely useless). But I left them here to keep the diff as minimal as possible.

**Review by StephenButtolph — COMMENTED** [review]
Listing a bunch of "notable" changes.

File: vms/avm/vm.go [hunk 18]
```diff
@@ -162,7 +152,7 @@ func (vm *VM) Initialize(
 	genesisBytes []byte,
 	_ []byte,
 	configBytes []byte,
-	toEngine chan<- common.Message,
+	_ chan<- common.Message,
```
Comment (coffeeavax) [inline vms/avm/vm.go:155]: if we decide to remove this `toEngine` channel, should we also remove `toEnginee` as part of `func (vm *VM) Linearize(_ context.Context, stopVertexID ids.ID, toEngine chan<- common.Message) error`

**Review by coffeeavax — APPROVED** [review]

File: vms/avm/vm.go (hunk 18 above)
Comment (StephenButtolph) [reply vms/avm/vm.go:155]: No, the `toEngine` channel in `Linearize` is used to communicate to the `Snowman` consensus engine.

File: vms/avm/index_test.go [hunk 19]
```diff
@@ -5,349 +5,177 @@ package avm
...
-	}
+	assertIndexedTX(t, env.vm.db, 0, addr, txAssetID.ID, tx.ID())
+	assertLatestIdx(t, env.vm.db, addr, txAssetID.ID, 1)
 }
 
 func TestIndexer_Read(t *testing.T) {
 	require := require.New(t)
 
-	// setup vm, db etc
-	_, vm, _, _, _ := setup(t, true)
```
Comment (hexfusion) [inline vms/avm/index_test.go:330]: :tada:

File: vms/avm/vm_benchmark_test.go [hunk 20]
```diff
@@ -52,7 +55,7 @@ func BenchmarkLoadUser(b *testing.B) {
 		require.NoError(user.Close())
 	}
 
-	benchmarkSize := []int{10, 100, 1000, 10000}
+	benchmarkSize := []int{10, 100, 1000, 5000}
```
Comment (hexfusion) [inline vms/avm/vm_benchmark_test.go:58]: reasonable change

File: vms/avm/vm_test.go [hunk 21]
```diff
@@ -1032,296 +462,239 @@ func TestTxCached(t *testing.T) {
...
-	require.Len(txs, 1)
+	key := keys[0]
+	firstTx := &txs.Tx{Unsigned: &txs.BaseTx{
+		BaseTx: avax.BaseTx{
+			NetworkID:    constants.UnitTestID,
+			BlockchainID: chainID,
+			Ins: []*avax.TransferableInput{{
+				UTXOID: avax.UTXOID{
+					TxID:        env.genesisTx.ID(),
+					OutputIndex: 2,
```
Comment (hexfusion) [inline vms/avm/vm_test.go:515]: why is `OutputIndex` 2 not completely apparent from previous test. Looking at Outs I only see [0]

File: vms/avm/vm_test.go [hunk 22]
```diff
@@ -1473,34 +799,47 @@ func TestForceAcceptImportTx(t *testing.T) {
...
 		}},
 		SourceChain: constants.PlatformChainID,
 		ImportedIns: []*avax.TransferableInput{{
 			UTXOID: utxoID,
-			Asset:  avax.Asset{ID: genesisTx.ID()},
+			Asset:  txAssetID,
 			In: &secp256k1fx.TransferInput{
-				Amt:   1000,
-				Input: secp256k1fx.Input{SigIndices: []uint32{0}},
+				Amt: 1010,
```
Comment (hexfusion) [inline vms/avm/vm_test.go:823]: Why change amt for test?

File: vms/avm/vm_benchmark_test.go (hunk 20 above)
Comment (StephenButtolph) [reply vms/avm/vm_benchmark_test.go:58]: The maximum amount allowed here is actually `5000`. The benchmark for `10000` was failing on dev.

File: vms/avm/vm_test.go (hunk 21 above)
Comment (StephenButtolph) [reply vms/avm/vm_test.go:515]: This was just copied over from `setupIssueTx`. Any value in [0, 1, 2] would work here. The `OutputIndex` here is referring the the output from the tx it is consuming (`env.genesisTx`).

File: vms/avm/vm_test.go (hunk 22 above)
Comment (StephenButtolph) [reply vms/avm/vm_test.go:823]: This was just to align with the `TestIssueImportTx` function. Technically didn't need to be changed.

**Review by gyuho — APPROVED** [review]

**Pull request merged**